tests/framework/e2e/etcdctl.go

```diff
@@ -39,6 +41,12 @@ func NewEtcdctl(cfg *EtcdProcessClusterConfig, endpoints []string) *EtcdctlV3 {
 	}
 }
 
+func (ctl *EtcdctlV3) WithAuth(username, password string) *EtcdctlV3 {
```

Comment:
This doesn't return new client, but mutates original client and returns it.

Reply:
It follows a `NewEtcdctlV3`. So I think it should be fine.

Reply:
> It follows a `NewEtcdctlV3`. So I think it should be fine.

What do you mean?

Reply:
I meant `WithAuth` follows `NewEtcdctlV3`. So it should be fine that it is not mutating existing client. 

However, I accept your suggestion to pass a option to `Client()` instead. So this discussion around `WithAuth` can be dropped. 

```
func (c *e2eCluster) Client() Client {
	return e2eClient{e2e.NewEtcdctl(c.Cfg, c.EndpointsV3())}
}

func (c *e2eCluster) ClientWithAuth(userName, password string) Client {
	return e2eClient{e2e.NewEtcdctl(c.Cfg, c.EndpointsV3()).WithAuth(userName, password)}
}
```